### PR TITLE
[Outlook] Add Session Data APIs

### PR DESCRIPTION
**Pull request by danielebercovici**
Add Session Data APIs:
getAsync, setAsync, getAllAsync. clearAsync and removeAsync

### CODE REVIEW

**Comment by ElizabethSamuel-MSFT** [comment]
Not ready to be merged yet.

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -17444,6 +17444,116 @@ declare namespace Office {
...
+         * @param options - Optional. An object literal that contains one or more of the following properties.
+         *        `asyncContext`: Developers can provide any object they wish to access in the callback method.
+         * @param callback - Optional. When the method completes, the function passed in the `callback` parameter is called with a single parameter,
+         *                `asyncResult`, which is an `Office.AsyncResult` object.
+         *
+         * @beta
+         */
+        removeAsync(name:string, options?: Office.AsyncContextOptions, callback?: (asyncResult: Office.AsyncResult<void>) => void): void;
+        /**
+         * Get all session data key-value pairs.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:17521]:
```suggestion
         * Gets all session data key-value pairs.
```

**Review by ElizabethSamuel-MSFT — COMMENTED** [review]
@danielebercovici My initial review...

**Comment by rbuckton** [comment]
@ElizabethSamuel-MSFT Is this ready to be merged, or should I convert the PR to a draft?

**Comment by ElizabethSamuel-MSFT** [comment]
@rbuckton I'll be dealing with this tomorrow. Thanks for asking.

**Pull request merged**